### DISCUSSION TITLE
Longer names or description field for Logical Switches, Special Functions, and Global functions

### DISCUSSION THREAD

**Discussion by Dweinbach** [EdgeTX/edgetx#1075] [Ideas]
One thing that is hard in OpenTx and EdgeTX is keeping track of what these various special and global functions and logical switches are programmed to do in each model.

It would be nice to be able to use longer names at least for these, or also for telemetry values, channel names, etc.

I realize that it might be hard to change these as they are buried deep and throughout the code, so an alternative would be to add a "friendly name" field or an "alias" that would be used when selecting one of these objects, even if the resulting equation only stored the "actual", original short name.

**Comment by raphaelcoeffic** [comment]
In fact, it’s not hard at all anymore thx to YAML. However, it will cost some memory, so that I would prefer to have this only on targets where we have plenty of memory, so those with color screen.

**Comment by ChrisOhara57** [comment]
I tried some LUA programming in ETHOS simulator (test conversion of two existing scripts from OTX).
What I really liked is the fact that the logical switches do have names.
So in Lua you can address the logical switches as source by name and you do not have to found out their "magic number" (which is hardware dependent).
This makes it much easier to make scripts user and hardware independent.  (in OTX you have to make sure that the logical switches you use in scripts are not used for something else or another script)

**Comment by JimB40** [comment]
And we have B%W tagets legacy that will make a need to revamp every single B&W target UI
